### ISSUE TITLE
developers api is not working

### ISSUE THREAD

**Issue by farzan3** [facundoolano/google-play-scraper#284]
api is not working with developer name .. i think again google changed template .
i check parseList.js and i am sure template is changed . 
I am c# developer i wish i can help with nodejs :)

Thanks for any kind of help 

* Operating System:
* Node version:
* google-play-scraper version:

### Description: 
Describe your issue here

### Example code:
```js
// Put code to reproduce the issue here
```

### Error message:

```
Put error message here
```

**Comment by vitaliy901** [comment]
For me not working too :(

**Comment by vitaliy901** [comment]
Maybe someone gotta write a fix ? Please guys

**Comment by Felixaep** [comment]
Another one that does not work

**Comment by farzan3** [comment]
please fix it

**Comment by facundoolano** [comment]
uhm I don't know folks, maybe if you send another ten comments here and keep asking in twitter I'll pause the rest of my life and fix your issue.

**Comment by vitaliy901** [comment]
> uhm I don't know folks, maybe if you send another ten comments here and keep asking in twitter I'll pause the rest of my life and fix your issue.

sorry, bro )

**Comment by farzan3** [comment]
hey guys i can fix it :D but that will not be professional 🗡

**Comment by vitaliy901** [comment]
> hey guys i can fix it :D but that will not be professional 🗡

Ok, send it in Pull Request, but keep Object keys the same )

**Comment by vitaliy901** [comment]
> hey guys i can fix it :D but that will not be professional 🗡

it will be very helpfull )

**Comment by farzan3** [comment]
> > hey guys i can fix it :D but that will not be professional 🗡
> 
> it will be very helpfull )

just did that . its working for me :) i am c# dev not nodejs 👯‍♂️

**Comment by vitaliy901** [comment]
> done dude

thanks )

**Comment by dsharkov** [comment]
Guys, do you have any updates on this issue?
Have the same issue - call "gplay.developer({devId: "Netmarble"}).then(console.log);" and get an empty collection((

**Comment by Felixaep** [comment]
Hi, I tryed to fixed this ... But i need change de devId by developerId and now this parameter is a number:
example: https://play.google.com/store/apps/dev?id=6577204690045492686 

this is the changes for developer.js

`'use strict';

const request = require('./utils/request');
const cheerio = require('cheerio');

function getClp (html) {
  // Try to find clp from "next page" html elem.
  let match = html.match(/\?clp=(.*?)">/);
  // ... if we don't have it, we're probably on innerPage;
  // try to parse it from search_collection_more_results_cluster instead
  // var curl='https://play.google.com/store/apps/collection/search_collection_more_results_cluster?clp\x3dggENCgVwYW5kYRABGgIIAA%3D%3D:S:ANO1ljKV8KM';
  if (!match) match = html.match(/\?clp\\x3d(.*?)';/);
  return match && match[1].replace(/%3D/g, '=');
}

function getNextToken (html) {
  // extract the token for the next page request
  // for the record, I hate regexps
  // const s = html.match(/\\42(GAE.+?)\\42/);
  const s = html.match(/\\x22-p6(.*?):S:(.*?)\\x22/g);
  if (!s) {
    return undefined;
  }
  return s[0].replace(/\\\\u003d/g, '=').replace(/\\x22/g, '');
}

/*
 * Extract navigation tokens for next pages, parse results and call
 * `checkFinished` to repeat the process with next page if necessary.
 */
function processAndRecur (html, opts, savedApps, clp) {
  const nextToken = getNextToken(html);
  clp = clp || getClp(html);

  const $ = cheerio.load(html);
  return opts.getParseList(opts)($)
    .then((newApps) => savedApps.concat(newApps))
    .then((apps) => checkFinished(opts, apps, nextToken, clp));
}

/*
 * If already have requested results or there are no more pages, return current
 * app list, otherwise request the ajax endpoint of the next page and process
 * the results.
 */
function checkFinished (opts, savedApps, nextToken, clp) {
  if (savedApps.length >= opts.num || !nextToken) {
    return savedApps.slice(0, opts.num);
  }

  const requestOptions = Object.assign({
    url: 'https://play.google.com/store/apps/collection/search_results_cluster_apps',
    method: 'POST',
    form: {
      num: savedApps.length === 49 ? 0 : 48, // confirm if always 48 works
      start: savedApps.length - 49,
      pagTok: nextToken,
      clp,
      pagtt: 3,
      hl: opts.lang,
      gl: opts.country
    },
    // we need an unescaped qs for the requests to work
    qsStringifyOptions: {encode: false},
    followAllRedirects: true
  }, opts.requestOptions);

  return request(requestOptions, opts.throttle)
    .then((html) => processAndRecur(html, opts, savedApps, clp));
}

/*
 * Make the first search request as in the browser and call `checkfinished` to
 * process the next pages.
 */
function initialRequest (opts) {
  // sometimes the first result page is a cluster of subsections,
  // need to skip to the full results page
  function seeMore (html) {
    const clp =  getClp(html);
    if (clp) {
      const innerUrl = `https://play.google.com/store/apps/collection/cluster?clp=${clp}&hl=${opts.lang}&gl=${opts.country}` ;
      return request(Object.assign({
        url: innerUrl
      }, opts.requestOptions), opts.throttle);
    }
    return html;
  }
  function skipClusterPage (html) {
    const match = html.match(/href="\/store\/apps\/collection\/search_collection_more_results_cluster?(.*?)"/);
    if (match) {
      const innerUrl = 'https://play.google.com/' + match[0].split(/"/)[1];
      return request(Object.assign({
        url: innerUrl
      }, opts.requestOptions), opts.throttle);
    }
    return html;
  }

  const url = `https://play.google.com/store/apps/dev?id=${opts.developerId}`;
  return request(Object.assign({url}, opts.requestOptions), opts.throttle)
    .then(seeMore)
    .then(skipClusterPage)
    .then((html) => processAndRecur(html, opts, []));
}

function getPriceGoogleValue (value) {
  switch (value.toLowerCase()) {
    case 'free':
      return 1;
    case 'paid':
      return 2;
    case 'all':
    default:
      return 0;
  }
}

function developer (getParseList, opts) {
  return new Promise(function (resolve, reject) {
    if (!opts || !opts.developerId) {
      throw Error('Search term missing');
    }

    if (opts.num && opts.num > 250) {
      throw Error("The number of results can't exceed 250");
    }

    opts = {
      developerId: encodeURIComponent(opts.developerId),
      lang: opts.lang || 'en',
      country: opts.country || 'us',
      num: opts.num || 20,
      fullDetail: opts.fullDetail,
      price: opts.price ? getPriceGoogleValue(opts.price) : 0,
      throttle: opts.throttle,
      cache: opts.cache,
      getParseList,
      requestOptions: opts.requestOptions
    };

    initialRequest(opts)
      .then(resolve)
      .catch(reject);
  });
}


module.exports = developer;
`

**Comment by tdunham02** [comment]
Google has changed this page and now the data is in a script, similar to the app's page. You'll have the parse the script tags and I believe it's `ds:3` and you'll map out the keys for each piece of metadata. Hope this helps.

**Comment by facundoolano** [comment]
@hunnapikes if you have the time, can you take a look to see if the code you added for search/similar can be used for developers as @tdunham02 suggests?

**Comment by tdunham02** [comment]
@facundoolano @hunnapikes The search results page has also been updated to the script tags in `ds:3`. Have to look back at how to get the `pagTok` for the next page of results though...

**Comment by facundoolano** [comment]
@tdunham02 yeah, @hunnapikes has been working already on search and similar to parse the new script tags. I _think_ pagination is also working there, although not sure.

**Comment by tdunham02** [comment]
@facundoolano Oh ok! I'll take a look at it and check out the pagination.

**Comment by MojtabaRazmi** [comment]
> hey guys i can fix it :D but that will not be professional 🗡

i can't see your code!
where is it?

**Comment by chengchengkang** [comment]
Hi, what I did is (using python):
- I have a list of app devloperID
- Use the googleplay store this url combine with developerID :"https://play.google.com/store/apps/developer?id={}"
- and then use bs4 to get all App ID(s) showing on the page
- use google-play-scraper to get App details

This is my way as I can think the easiest way to solve this problem temporarily. 
Hope this helps :)

**Comment by Suvitruf** [comment]
Any updates?

**Comment by Suvitruf** [comment]
Ok, I guess, I can make it works.

1st of all, there different types for dev page right now:
1) With string id, e.g. https://play.google.com/store/apps/developer?id=Google LLC
2) Int id, e.g. https://play.google.com/store/apps/dev?id=5700313618786177705

For 2nd one I've found work around. For 1st one...it works mostly like `search` request. I can receive 1st batch with `batchexecute`, but couldn't find token to request next batch )=

**Comment by Suvitruf** [comment]
I've created PR. Not sure if it's the optimal way to do it. At least, it works now (:

**Comment by KemikalGeneral** [comment]
Hi guys!

Just had a message from the manager of a project I worked on last year that uses this package, asking if I'd take a look as it isn't working anymore.

Saw this thread and got worried... Is it working now?


Thanks,

**Comment by facundoolano** [comment]
fixed by #344
